### DISCUSSION TITLE
What's the proper way of upgrading to a WebSocket?

### DISCUSSION THREAD

**Discussion by gcocatre** [oakserver/oak#308] [Q&A] [answered]
Hello,

I'm running a rudimentary chat service and I use Oak for both serving static files, and upgrading to a WebSocket for real time communication.

However I see lots of entries in the logs about 404 errors on my WebSocket route. I.e. "404 GET /web-socket/lobby - 60025ms". What am I doing wrong, or not doing? Here's the relevant router block:

```
router.get("/web-socket/:room", (context) => {
  const req = context.request;
  const room = context!.params!.room || "default";
  if (context.socket) {
    return chat(context.socket, req, room).catch((e) => console.error(e));
  } else {
    return context
      .upgrade()
      .then((socket) => chat(socket, req, room))
      .catch((e) => console.error(e));
  }
});
```

**Comment by Hunam6** [comment]
It may be due to this
![image](https://user-images.githubusercontent.com/38606542/117557554-2a75f300-b074-11eb-8137-87ee333a45cc.png)

**Comment by gcocatre** [comment, marked as answer]
Actually I figured it out. I was using a logger middleware that should ignore requests for which `context.respond` was set to false.
